torch/csrc/autograd/init.cpp

```diff
@@ -250,6 +252,40 @@ PyObject* THPAutograd_initExtension(PyObject* _unused, PyObject *unused) {
 #endif
   });
 
+  // WARNING: These functions are not operators and may not behave as expected
+  // in TorchScript tracing, FX transforms, or operator serialization.
+  // Creates a new profiling scope using RecordFunction and invokes its starting
+  // callbacks.
+  m.def("record_function_with_args_enter", [](const std::string& name, py::args args) {
```

Comment:
Are these expected to be public API?

Reply:
@albanD I think this will be useful in many contexts, but I know introducing a new public API needs some careful thoughts. I'm happy to make it private with a `_` prefix: `_record_function_with_args_enter` for now, so we can start using them internally and if we feel more confident in the future on its usage and capability, we can formalize it to a public API.